src/__tests__/links.integration.test.js

```diff
@@ -0,0 +1,40 @@
+/* eslint-disable complexity */
+const fs = require('fs');
+const path = require('path');
+const { JSDOM } = require('jsdom');
+
+function extractAnchors(filePath, urlPath) {
+  const fileContents = fs.readFileSync(filePath, 'utf-8');
+  const dom = new JSDOM(fileContents);
+  const links = Array.from(dom.window.document.querySelectorAll('a'))
+    .map(link => link.href)
+    .filter(href => href && href.startsWith('about:blank#') && href !== 'about:blank#')
+    .map(href => `${urlPath}${href.split('about:blank')[1]}`);
+  return links;
+}
+
+function extractLinksRecursively(directory) {
```

Comment:
@AndyClifford Change: These functions should be utilities and have their own tests